### ISSUE TITLE
[BF16]For GPT2ForSequenceClassification model on stock pytorch, Softmax cost time on pvc-1100 3x of A100

### ISSUE THREAD

**Issue by xiaowangintel** [pytorch/pytorch#134070]
### 🐛 Describe the bug

pvc-1100 perf data

-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
                                                   Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg      Self XPU    Self XPU %     XPU total  XPU time avg    # of Calls
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
                                         aten::_softmax        13.07%       8.951ms        23.80%      16.306ms     135.885us      67.493ms       100.00%      67.493ms     562.441us           120
at::native::xpu::impl::DispatchSoftmaxForwardKernelF...         0.00%       0.000us         0.00%       0.000us       0.000us      67.493ms       100.00%      67.493ms     562.441us           120
                                          aten::softmax         0.11%      74.286us        23.91%      16.380ms     136.504us       0.000us         0.00%      67.493ms     562.441us           120
                                        piDeviceGetInfo         4.33%       2.968ms         4.33%       2.968ms       0.343us       0.000us         0.00%       0.000us       0.000us          8642
                                piextDeviceSelectBinary         0.90%     614.818us         0.90%     614.818us       0.320us       0.000us         0.00%       0.000us       0.000us          1920
                                        piProgramRetain         0.20%     134.186us         0.20%     134.186us       0.280us       0.000us         0.00%       0.000us       0.000us           480
                                         piKernelRetain         0.20%     136.735us         0.20%     136.735us       0.285us       0.000us         0.00%       0.000us       0.000us           480
                                   piKernelGetGroupInfo         0.24%     167.326us         0.24%     167.326us       0.697us       0.000us         0.00%       0.000us       0.000us           240
                                        piKernelRelease         0.15%     100.487us         0.15%     100.487us       0.279us       0.000us         0.00%       0.000us       0.000us           360
                                       piProgramRelease         0.15%     100.771us         0.15%     100.771us       0.280us       0.000us         0.00%       0.000us       0.000us           360
                               piextKernelSetArgPointer         0.24%     164.040us         0.24%     164.040us       0.684us       0.000us         0.00%       0.000us       0.000us           240
                                         piKernelSetArg         0.94%     643.418us         0.94%     643.418us       0.536us       0.000us         0.00%       0.000us       0.000us          1200
                                  piEnqueueKernelLaunch         3.34%       2.290ms         3.34%       2.290ms      19.080us       0.000us         0.00%       0.000us       0.000us           120
                                         piEventRelease         0.05%      36.058us         0.05%      36.058us       0.300us       0.000us         0.00%       0.000us       0.000us           120
                                          piQueueFinish        76.09%      52.123ms        76.09%      52.123ms     407.213us       0.000us         0.00%       0.000us       0.000us           128
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
Self CPU time total: 68.504ms
Self XPU time total: 67.493ms

A100 pref data

(xw_1) sparse@mlp-milan-01:/mnt/dataset/xw/inductor_log/huggingface/GPT2ForSequenceClassification/inference/bfloat16/18$ python test_softmax.py
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
                                                   Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg     Self CUDA   Self CUDA %    CUDA total  CUDA time avg    # of Calls
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
                                         aten::_softmax         6.95%       1.652ms         8.88%       2.110ms      17.582us      22.906ms       100.00%      22.906ms     190.883us           120
void (anonymous namespace)::softmax_warp_forward<c10...         0.00%       0.000us         0.00%       0.000us       0.000us      22.906ms       100.00%      22.906ms     190.883us           120
                                          aten::softmax         0.39%      93.820us         9.27%       2.204ms      18.363us       0.000us         0.00%      22.906ms     190.883us           120
                                       cudaLaunchKernel         1.92%     457.311us         1.92%     457.311us       3.811us       0.000us         0.00%       0.000us       0.000us           120
                                  cudaDeviceSynchronize        90.73%      21.564ms        90.73%      21.564ms      10.782ms       0.000us         0.00%       0.000us       0.000us             2
-------------------------------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------  ------------
Self CPU time total: 23.767ms
Self CUDA time total: 22.906ms

 

 

UT test:

import torch
import contextlib

input = torch.randn([4, 12, 1024, 1024], dtype = torch.bfloat16).to("xpu")

prof_null = contextlib.nullcontext()
prof_xpu = torch.profiler.profile(
    activities=[
    torch.profiler.ProfilerActivity.CPU,
    torch.profiler.ProfilerActivity.XPU],
)

for i in range(10):
    if i == 9:
        prof = prof_xpu
    else:
        prof = prof_null
    with prof:
        for i in range(120):
            b = torch.nn.functional.softmax(input, dim=-1)
        torch.xpu.synchronize()
    if hasattr(prof, "export_chrome_trace"):
        print(prof.key_averages().table(sort_by="self_xpu_time_total", row_limit=100000))

### Versions

pytorch commit:03480213dea1f60f6d12e7348904d2f3ef7314d0
torch-xpu-ops commit:718bc42c667539977e5eadb11ea4dec602544bf2
driver : hotfix_agama-ci-devel-881.19
pti : l_intel-pti-dev_p_0.9.0.38_offline.sh
basekit : l_BaseKit_p_2024.2.1.100_offline.sh

cc @msaroufim @gujinghui @EikanWang @fengyuan14 @guangyey

**Comment by EikanWang** [comment]
@xiaowangintel , thanks for the information. The PVC 1100 is a single-tile card. But the CPU overhead, it seems like, does not make sense.

**Comment by xiaowangintel** [comment]
This is an xpu operator issue. And I will remove it.
